### PR TITLE
fixes #19: add merge operation

### LINKED ISSUE #19: add merge operation
to CrudOperations
to Translator.kt: line 75
and AugmentationTest

only if the type has an ID to merge on.

### PR DESCRIPTION
**Pull request by conker84**
Fixes #19 

Add merge operation only if the type has an ID to merge on.

### CODE REVIEW

File: src/test/resources/movie-test.md [hunk 1]
```diff
@@ -339,3 +339,23 @@ RETURN movie { .title,actors:[(movie)<-[:ACTED_IN]-(movieActors:Actor) |
...
+  }
+}
+```
+
+```params
+{"actorUserId": "1", "actorName": "Andrea"}
+```
+
+```cypher
+MERGE (actor:Actor {userId:$actorUserId}) ON CREATE SET actor.userId=$actorUserId,actor.name=$actorName
```
Comment (jexp) [inline src/test/resources/movie-test.md]: we should use SET not 
ON CREATE SET

**and not set the id again**

File: src/test/resources/movies-test-schema.graphql [hunk 2]
```diff
@@ -1,5 +1,5 @@
 type Movie {
-  _id: String
+#  _id: String
```
Comment (jexp) [inline src/test/resources/movies-test-schema.graphql]: can you add an issue for the _id field ?

File: src/test/resources/movies-test-schema.graphql [hunk 3]
```diff
@@ -75,14 +75,18 @@ type Book {
...
   MovieById(movieId: ID!): Movie
-  MovieBy_Id(_id: String!): Movie
   GenresBySubstring(substring: String): [Genre] @cypher(statement: "MATCH (g:Genre) WHERE toLower(g.name) CONTAINS toLower($substring) RETURN g")
   Books: [Book]
 }
 type Mutation {
   createGenre(name:String): Genre @cypher(statement:"CREATE (g:Genre) SET g.name = name RETURN g")
 }
 # scalar DateTime
+
```
Comment (jexp) [inline src/test/resources/movies-test-schema.graphql]: can you also add an issue for this one?

File: src/main/kotlin/org/neo4j/graphql/Translator.kt [hunk 4]
```diff
@@ -66,9 +66,15 @@ class Translator(val schema: GraphQLSchema) {
                 // todo extract method or better object
                 val properties = properties(variable, fieldDefinition, propertyArguments(field))
```
Comment (jexp) [inline src/main/kotlin/org/neo4j/graphql/Translator.kt]: we can separate out all-properties and properties without-id which we can use later on in SET, UPDATE etc.

Comment (conker84) [reply src/main/kotlin/org/neo4j/graphql/Translator.kt]: Is it good for you if we manage that in the `setProperties` method by providing a list of properties to exclude?

File: src/test/resources/movie-test.md (hunk 1 above)
Comment (conker84) [reply src/test/resources/movie-test.md]: ok

File: src/test/resources/movies-test-schema.graphql (hunk 3 above)
Comment (conker84) [reply src/test/resources/movies-test-schema.graphql]: I restored the old version and with the actual changes all works, as well as for the _id field so we don't need to open an issue for them

**Comment by conker84** [comment]
Thanks a lot for the feedback!
I updated the issue

File: src/main/kotlin/org/neo4j/graphql/Translator.kt (hunk 4 above)
Comment (jexp) [reply src/main/kotlin/org/neo4j/graphql/Translator.kt]: that's fine to, but then we should rename the method :)

**Comment by jexp** [comment]
Thanks a lot.

**Pull request merged**